server/eclipse-project/src/main/java/us/freeandfair/corla/model/CountyDashboard.java

```diff
@@ -405,32 +412,52 @@ public AuditBoard currentAuditBoard() {
    * 
    * @param the_members The members.
    */
-  public void signInAuditBoard(final List<Elector> the_members) {
-    if (my_current_audit_board_index == null) {
-      my_current_audit_board_index = my_audit_boards.size();
-    } else {
-      final AuditBoard current = my_audit_boards.get(my_current_audit_board_index);
-      current.setSignOutTime(Instant.now());
-      my_current_audit_board_index = my_current_audit_board_index + 1;
+  public void signInAuditBoard(final Integer index,
+                               final List<Elector> the_members) {
+    final AuditBoard currentBoard = my_audit_boards.get(index);
+    final AuditBoard newBoard = new AuditBoard(the_members, Instant.now());
+
+    if (currentBoard != null) {
+      this.signOutAuditBoard(index);
     }
-    my_audit_boards.add(new AuditBoard(the_members, Instant.now()));
+
+    my_audit_boards.put(index, newBoard);
   }
-  
+
   /**
    * Signs out the current audit board.
-   * 
-   * @exception IllegalStateException if no audit board is signed in.
+   *
+   * If no audit board is present at the given index, nothing is changed.
    */
-  public void signOutAuditBoard() {
-    if (my_current_audit_board_index == null) {
-      throw new IllegalArgumentException("no audit board signed in");
```

Comment:
This would change behavior a little bit, right? The exception would have been caught elsewhere and a `500` returned, but now we'll get a `200` from a no-op. Seems reasonable to me, but I'm trying to think if this change would have unexpected effects.

Reply:
Yeah, it's changing. There are situations where the client might have an older idea about what audit board is signed in, sends a message signing out a given audit board, but by the time the message gets here, the board is already signed out (or maybe they clicked twice on something, etc.). It seems a bit more likely to run into this kind of thing with the multiple audit board UI that's coming up, so I'd rather allow it and let the client update its own state or whatever than throw a potentially spurious exception.